### ISSUE TITLE
appl_getinfo_str() returns empty string in release builds (and others bugs)

### ISSUE THREAD

**Issue by xdelatour** [freemint/freemint#293]
It seems appl_getinfo() has multiple flaws
- appl_getinfo_str(AES_VERSION) returns ap_gout3 only [if XaAES is not build as Release](https://github.com/freemint/freemint/blob/8ea9d1ab63b7a56ebc9ed9e65c7d1254733fdc67/xaaes/src.km/info.c#L31): Alpha and Beta are OK but Release returns an empty string (I guess XaAES is always build as Beta but I wanted to confirm this behaviour)
- appl_getinfo(AES_LARGEFONT) and appl_getinfo(AES_SMALLFONT) return infos about fonts defined [at startup only](https://github.com/freemint/freemint/blob/2f027a2dbbc306b1bb628397307f67c69295d840/xaaes/src.km/k_init.c#L866) instead of current font, except the large font height which is [updated for each appl_getinfo() call](https://github.com/freemint/freemint/blob/8ea9d1ab63b7a56ebc9ed9e65c7d1254733fdc67/xaaes/src.km/xa_appl.c#L1568) (it means appl_getinfo may returns the right height and the wrong font ID?). In XaAES, the AES standard font is the font defined by any AES client who sets the menu bar.
- wing_get(WF_XAAES) always [returns 9](https://github.com/freemint/freemint/blob/8ea9d1ab63b7a56ebc9ed9e65c7d1254733fdc67/xaaes/src.km/xa_wind.c#L1747) ([HEX_VERSION=$09](https://github.com/freemint/freemint/blob/8ea9d1ab63b7a56ebc9ed9e65c7d1254733fdc67/xaaes/src.km/version.h#L41-L42)) which has no meaning: the right value should be the XaAES version number. According to tos.hyp, version looks like $0964 and wi_ghandle must be set to zero before calling: it's wrong too, wi_gw1 is set to 9 for any value of wi_ghandle

**Comment by th-otto** [comment]
for the version: i think it doesn't make sense to report alpha and beta for a release version, so the current behaviour seems ok to me, since there is no release version to report.

for the font: does this ever change during runtime? 

for WF_XAAES: yes, the value 0x09 seems bogus. But the comment also indicates that this was about to be removed completely.

**Comment by xdelatour** [comment]
The release version ([if DEV_STATUS is set to release](https://github.com/freemint/freemint/blob/6bed25f61ef99023d0633053cb8720839494ebf3/xaaes/src.km/version.h#L37-L39)) reports nothing because the info_string is left empty.

My bad. Indeed font ID is never updated.

About WF_XAAES, there is no info in newcalls.txt either. So probably to remove from tos.hyp and XaAES.

**Comment by mikrosk** [comment]
Ok, so this looks like separate issues to address:

- reporting a real version if DEV_STATUS is set to release
- font size reporting seems ok, so no action needed
- removal of WF_XAAES from tos.hyp and xaaes source code

Correct?

**Comment by xdelatour** [comment]
- Yes, IMHO stable release must return a valid string too. It implies About dialog must be updated (details about compiler are not need for stable release, it's why XaAES creates an empty string)
- Right
- and removal from gemlib (mt_gem.h.in) with potential sides effects on applications that use it

**Comment by xdelatour** [comment]
Patch to always get a infostring from appl_getinf_str:

[remove_info.patch.txt](https://github.com/freemint/freemint/files/11299492/remove_info.patch.txt)

- The infostring is now a string literal
- The few lines about fonts are put back to k_init
- info.c is removed
- WF_XAAES is not removed because there is a pending patch somewhere to add WF_INFOXYWH (require changes to xa_wind.c)

**Comment by xdelatour** [comment]
I agree with Thorsten, we shouldn't implement WF_M_WINDLIST and others WINX features (WF_STACK, WF_TOPALL, WF_BOTTOMALL...) which probably rely on the same window list.

XaAES has its own functions to manage the window list, with bugs as noticed by @GokMasE (#266), but emulating WF_M_WINDLIST would add a lot of overload.

On the contrary, 
- WF_INFOXYWH is trivial to implement (despite XaAES widget are not AES object)
- wind_set(WF_KIND) may be useful and is supported by both MagiC and MyAES. I believe I'm able to add it but with a lot of effort (the function already exists but a lot of checks are required because of toolbar and window-menu)
- WF_WIDGETS would be great, N.AES handles it successfully with 'old' GEM indices of AES object (10,11,15,16 as up,dn,lf,rt arrows) and its totally different RSC-file. (I guess XaAES wanted to do the same with the `GtoX` array). The problem though is W_HBAR/W_VBAR-equivalents have more than three components (sliders have two uparrows and two leftarrows).  

It would be nice to work on fVDI too. I noticed yesterday vq_px_format() always return $3421820, whatever is the video mode, including 8 bits ($3421820 is PX_xFIRST + PX_PACKED + PX_3COMP + PX_USES24 + PX_32BIT).

**Comment by xdelatour** [comment]
I think we can leave things as they are for now and forget this patch, but in this case we need to update tos.hyp to specify that info_string can be empty. 

Only [comments in info.c](https://github.com/freemint/freemint/blob/9a1f35a60984ff41bc868a1dd54e0a51de85f385/xaaes/src.km/info.c#L1-L6) say info_string is available if it's a non-release-version. 
IMHO this conflicts with [newcalls.txt](https://github.com/freemint/freemint/blob/9a1f35a60984ff41bc868a1dd54e0a51de85f385/xaaes/src.km/newcalls.txt#L145-L155) where it's not stated explicitly (possibly implied by 'development status'?).
